tanner/session_analyzer.py

```diff
@@ -33,16 +34,19 @@ def __init__(self, loop=None):
             s_key = session['snare_uuid']
             del_key = session['sess_uuid']
             try:
-                await redis_client.lpush(s_key, [json.dumps(session)])
-                await redis_client.delete([del_key])
-            except asyncio_redis.NotConnectedError as redis_error:
+                await redis_client.execute('lpush', s_key, *[json.dumps(session)])
+                await redis_client.execute('del', *[del_key])
+            except aioredis.ProtocolError as redis_error:
                 self.logger.error('Error with redis. Session will be returned to the queue: %s',
                                   redis_error)
                 self.queue.put(session)
 
     async def create_stats(self, session, redis_client):
         sess_duration = session['end_time'] - session['start_time']
-        rps = session['count'] / sess_duration
+        if sess_duration != 0:
+            rps = session['count'] / sess_duration
+        else:
+            rps = 0
         location_info = await self._loop.run_in_executor(
             None, self.find_location, session['peer']['ip']
```

Comment:
```
Traceback (most recent call last):
  File "/usr/local/lib64/python3.6/site-packages/aiohttp/web_protocol.py", line 422, in start
    resp = yield from self._request_handler(request)
  File "/usr/local/lib64/python3.6/site-packages/aiohttp/web.py", line 306, in _handle
    resp = yield from handler(request)
  File "/usr/local/lib/python3.6/site-packages/Tanner-0.5.0-py3.6.egg/tanner/server.py", line 60, in handle_event
    data, self.redis_client
  File "/usr/local/lib/python3.6/site-packages/Tanner-0.5.0-py3.6.egg/tanner/session_manager.py", line 17, in add_or_update_session
    await self.delete_old_sessions(redis_client)
  File "/usr/local/lib/python3.6/site-packages/Tanner-0.5.0-py3.6.egg/tanner/session_manager.py", line 71, in delete_old_sessions
    is_deleted = await self.delete_session(sess, redis_client)
  File "/usr/local/lib/python3.6/site-packages/Tanner-0.5.0-py3.6.egg/tanner/session_manager.py", line 90, in delete_session
    await self.analyzer.analyze(sess.get_uuid(), redis_client)
  File "/usr/local/lib/python3.6/site-packages/Tanner-0.5.0-py3.6.egg/tanner/session_analyzer.py", line 27, in analyze
    result = await self.create_stats(session, redis_client)
  File "/usr/local/lib/python3.6/site-packages/Tanner-0.5.0-py3.6.egg/tanner/session_analyzer.py", line 51, in create_stats
    None, self.find_location, session['peer']['ip']
  File "/usr/lib64/python3.6/concurrent/futures/thread.py", line 56, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/usr/local/lib/python3.6/site-packages/Tanner-0.5.0-py3.6.egg/tanner/session_analyzer.py", line 126, in find_location
    location = reader.city(ip)
  File "/usr/local/lib/python3.6/site-packages/geoip2/database.py", line 114, in city
    return self._model_for(geoip2.models.City, 'City', ip_address)
  File "/usr/local/lib/python3.6/site-packages/geoip2/database.py", line 195, in _model_for
    record = self._get(types, ip_address)
  File "/usr/local/lib/python3.6/site-packages/geoip2/database.py", line 188, in _get
    record = self._db_reader.get(ip_address)
  File "/usr/local/lib/python3.6/site-packages/maxminddb/reader.py", line 108, in get
    (string_type_name, type(ip_address).__name__))
TypeError: argument 1 must be str, not NoneType
```

Reply:
I have made changes, please check if this error occurs still. :)